src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -454,6 +478,86 @@ class IpexCommunicationService extends AgentService {
       notifRecord.a.r as NotificationRoute
     );
   }
+
+  @OnlineOnly
+  async grantAcdcById(credentialId: string) {
```

Comment:
remember next PR when we are hooking this up to an apply,offer,agree that we should update this function instead of creating a new one